core/templates/pages/exploration-player-page/services/exploration-engine.service.spec.ts

```diff
@@ -0,0 +1,909 @@
+    let initSuccessCb = jasmine.createSpy('success');
+
+    spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(false);
+
+    // Here 1 is default value, this is being initialized in the constructor.
+    expect(explorationEngineService.getExplorationVersion()).toBe(1);
+
+    explorationEngineService.init(
+      explorationDict, 2, null, true, ['en'], initSuccessCb);
+
+    const explorationVersion = explorationEngineService.getExplorationVersion();
+    expect(explorationVersion).toBe(2);
+  });
+
+  it('should return author recommended exploration Id\'s', () => {
+    let initSuccessCb = jasmine.createSpy('success');
+
+    spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(false);
+
+    expect(() => {
+      explorationEngineService.getAuthorRecommendedExpIds();
+    }).toThrowError(
+      'Cannot read property \'getAuthorRecommendedExpIds\' of undefined');
+
+    explorationEngineService.init(
+      explorationDict, 1, null, true, ['en'], initSuccessCb);
+
+    explorationEngineService.currentStateName = 'End';
+
+    const recommendedId = explorationEngineService.getAuthorRecommendedExpIds();
+    expect(recommendedId).toContain('recommnendedExplorationId');
+  });
+
+  it('should update current state when an answer is submitted ' +
+    'and a new card is recorded', () => {
+    let initSuccessCb = jasmine.createSpy('success');
+    let submitAnswerSuccessCb = jasmine.createSpy('success');
+    let answer = 'answer';
+    let answerClassificationResult = new AnswerClassificationResult(
+      outcomeObjectFactory.createFromBackendDict({
+        dest: 'Mid',
+        feedback: {
+          content_id: 'feedback_1',
+          html: 'Answer is correct!'
+        },
+        labelled_as_correct: true,
+        param_changes: [],
+        refresher_exploration_id: null,
+        missing_prerequisite_skill_id: null
+      }), 1, 0, 'default_outcome');
+
+    let lastCard = stateCardObjectFactory.createNewCard(
+      'Card 1', 'Content html', 'Interaction text', null,
+      null, null, 'content_id');
+
+    spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(false);
+    spyOn(playerTranscriptService, 'getLastStateName')
+      .and.returnValue('Start');
+    spyOn(playerTranscriptService, 'getLastCard').and.returnValue(lastCard);
+    spyOn(answerClassificationService, 'getMatchingClassificationResult')
+      .and.returnValue(answerClassificationResult);
+
+    expect(explorationEngineService.currentStateName).toBe(undefined);
+
+    explorationEngineService.init(
+      explorationDict, 1, null, true, ['en'], initSuccessCb);
+
+    explorationEngineService.submitAnswer(
+      answer, textInputService, submitAnswerSuccessCb);
+    expect(explorationEngineService.currentStateName).toBe('Start');
+    explorationEngineService.recordNewCardAdded();
+    expect(explorationEngineService.currentStateName).toBe('Mid');
+  });
+
+  it('should load initial state', () => {
```

Comment:
When should the initial state be loaded?

Reply:
Please make sure that the test descriptions generally, follow the format - ‘should do X if Y happens’. This and a couple of other places seems to be missing that. You can check under the `Test Descriptions` in the `Frontend Test - Review comments documentation` doc we have for more details.

Reply:
done!